helper/Datatypes.php

```diff
@@ -53,21 +53,35 @@ class Datatypes
      */
     public static function decodeFile($binary)
     {
-        
+
         $returnValue = ['name' => '', 'mime' => '', 'data' => ''];
-        
+
         if (empty($binary) === false) {
             $filenameLen = current(unpack('S', substr($binary, 0, 2)));
+            if ($filenameLen >= strlen($binary)) {
```

Comment:
This little piece of magic would be better off with an explicit comment on top IMO.
Just a suggestion, though.